### PR TITLE
Documentation build on CI

### PR DESCRIPTION
**Pull request by DanNixon**

### CODE REVIEW

File: .azure-pipelines/pull_request.yml [hunk 1]
```diff
@@ -64,12 +78,60 @@ stages:
...
+          - bash: conda install "$(packages_dir)/"scipp-*.tar.bz2
+            displayName: 'Install Scipp package'
+          - bash: |
+              set -ex
+              cd docs
+              sphinx-build . "$(docs_dir)"
+            displayName: 'Build documentation'
+          - task: PublishBuildArtifacts@1
+            inputs:
+              PathtoPublish: '$(docs_dir)'
```
Comment (SimonHeybrock) [inline .azure-pipelines/pull_request.yml]: Is this where it pushes to readthedocs?

Comment (SimonHeybrock) [reply .azure-pipelines/pull_request.yml]: If so, why is it in in `pull_request.yml`?

Comment (DanNixon) [reply .azure-pipelines/pull_request.yml]: No, this stores a copy in Azure artifacts or inspection.

**Comment by DanNixon** [comment]
> Do I understand correctly that this is essentially always making a conda package (even for PRs) which is then used to build the docs?

Yes, building Conda packages was always the plan since that is our main distribution means.

Hoover I advise not reviewing this yet as it is sill a work in progress and certainly will be changed further.

**Pull request closed without merging**